org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java

```diff
@@ -11,92 +11,177 @@
-            value = {NX_NUMBER}
-                @units = "mm"
-            controller_record = {NX_CHAR} //EPICS name
-        upstream_y:NXpositioner
-            name = "s1usY"
-            description = "Upstream Y position"
-            value = {NX_NUMBER}
-                @units = "mm"
-            controller_record = {NX_CHAR} //EPICS name
- </pre>
- 
- * 
- * @author Matthew Gerring
- *
+ * @see {@link MockScannableConfiguration}
  */
-public class MockNeXusSlit extends MockNeXusScannable {
+public class MockNeXusSlit extends MockScannable implements INexusDevice<NXslit> {
+
+	public boolean isWritingOn() {
+		return writingOn;
+	}
+
+	public void setWritingOn(boolean writingOn) {
+		this.writingOn = writingOn;
+	}
+
+	private ILazyWriteableDataset xLzSet;
+	private ILazyWriteableDataset yLzSet;
+	private ILazyWriteableDataset xLzValue;
+	private ILazyWriteableDataset yLzValue;
+
+	private boolean writingOn = true;
 
 	public MockNeXusSlit() {
 		super();
 	}
 
+	public MockNeXusSlit(String name, double d, int level) {
+		super(name, d, level);
+	}
 	public MockNeXusSlit(String name, double d, int level, String unit) {
 		super(name, d, level, unit);
 	}
 
-	public MockNeXusSlit(String name, double d, int level) {
-		super(name, d, level);
+	@ScanFinally
+	public void nullify() {
+		xLzSet   = null;
+		xLzValue = null;
+		yLzSet   = null;
+		yLzValue = null;
+	}
+
+	public NexusObjectProvider<NXslit> getNexusProvider(NexusScanInfo info) throws NexusException {
+		final NXslit positioner = NexusNodeFactory.createNXslit();
+
+		if (info.getScanRole(getName()) == ScanRole.METADATA) {
+			//positioner.setField(NXslit.NX_X_GAP, getPosition().doubleValue());
+			//positioner.setField(NXslit.NX_Y_GAP, getPosition().doubleValue());
+			positioner.setX_gapScalar(getPosition().doubleValue());
+			positioner.setY_gapScalar(getPosition().doubleValue());
+		} else {
+			String floatFill = System.getProperty("GDA/gda.nexus.floatfillvalue", "nan");
+			double fill = floatFill.equalsIgnoreCase("nan") ? Double.NaN : Double.parseDouble(floatFill);
+
+			xLzSet = positioner.initializeLazyDataset(NXslit.NX_X_GAP, 1, Double.class);
+			yLzSet = positioner.initializeLazyDataset(NXslit.NX_Y_GAP, 1, Double.class);
+			xLzSet.setFillValue(fill);
+			yLzSet.setFillValue(fill);
+			xLzSet.setChunking(new int[]{8}); // Faster than looking at the shape of the scan for this dimension because slow to iterate.
+			yLzSet.setChunking(new int[]{8}); // Faster than looking at the shape of the scan for this dimension because slow to iterate.
+			xLzSet.setWritingAsync(true);
+			yLzSet.setWritingAsync(true);
+
+			xLzValue = positioner.initializeLazyDataset(NXslit.NX_X_GAP, info.getRank(), Double.class);
+			yLzValue = positioner.initializeLazyDataset(NXslit.NX_Y_GAP, info.getRank(), Double.class);
+			xLzValue.setFillValue(fill);
+			yLzValue.setFillValue(fill);
+			xLzValue.setChunking(info.createChunk(false, 8)); // TODO Might be slow, need to check this

```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)